src/main/java/baseball/service/BaseBallGameService.java

```diff
@@ -0,0 +1,105 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+
+public class BaseBallGameService {
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final Validator validator;
+
+    public BaseBallGameService(InputView inputView, OutputView outputView, Validator validator) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.validator = validator;
+    }
+
+    public Player savePlayerNumber() {
+        return new Player(convertStringToNumber(requestPlayerNumber()));
+    }
+
+    private List<Integer> convertStringToNumber(String input) {
+        List<Integer> result = new ArrayList<>();
+        for (char c : input.toCharArray()) {
+            result.add(c - '0');
+        }
+        return result;
+    }
+
+    private String requestPlayerNumber() {
+        String input = inputView.printRequestInput();
+        validator.validate(input);
+        return input;
+    }
+
+    public void playGame() {
+        Computer computer = new Computer();
+        while(true) {
+            Player player = savePlayerNumber();
+            GameResult gameResult = numberComparator(computer, player);
+            Hint hint = generateHint(gameResult);
+            outputView.printHintMsg(hint);
+            if(gameResult.getStrike() == GameNumber.NUMBER_MAX_SIZE.getNumber()) {
+                outputView.printEndMsg();
+                break;
+            }
+        }
+    }
+    public boolean validRestartCommand(String restartCommand) {
+        if(Objects.equals(restartCommand, GameCommand.RESTART.getCommand())) {
+            return true;
+        } else if(Objects.equals(restartCommand, GameCommand.END.getCommand())) {
+            return false;
+        } else {
+            throw new IllegalArgumentException("1 또는 2를 입력하여 다시 시작하거나 종료해주세요 !");
+        }
+    }
+
+    public boolean gameIsEnd(GameResult gameResult) {
+        return gameResult.getStrike() == GameNumber.NUMBER_MAX_SIZE.getNumber();
+    }
+
+    public void startGame() {
+        outputView.printStartMsg();
+        while(true) {
+            playGame();
+            String restartCommand = inputView.gameRestartInputRequest();
+            if(validRestartCommand(restartCommand)) {
+                continue;
+            } else {
+                break;
+            }
+        }
+    }
+
+    public GameResult numberComparator(Computer computer, Player player) {
+        int strike = 0;
+        int ball = 0;
+        for(int i=0; i<computer.getComputerNumber().size(); i++) {
+            for(int j=0; j<player.getPlayerNumber().size(); j++) {
+                if(Objects.equals(computer.getComputerNumber().get(i), player.getPlayerNumber().get(j)) && i==j) {
+                    strike++;
+                } else if (Objects.equals(computer.getComputerNumber().get(i),player.getPlayerNumber().get(j)) && i!=j) {
+                    ball++;
+                }
+            }
+        }
```

Comment:
저라면 이 부분을 
```suggestion
         for(int i=0; i<computer.getComputerNumber().size(); i++) {
            if(Objects.equals(computer.getComputerNumber().get(i), player.getPlayerNumber().get(i))) {
                strike++;
            } else if (computer.getComputerNumber().contains(player.getPlayerNumber().get(i))) {
                ball++;
            }
        }
```
이렇게 작성해서 하나의 for문으로 간소화 했을거 같습니다.

저는 이번 과제하면서 
https://github.com/woowacourse/woowacourse-docs/blob/main/cleancode/pr_checklist.md 여기에 나와있는 클린 코드 원칙들을 준수하려고 노력하면서 작성했는데 그 중 한 메서드에 한 단계의 들여쓰기만 지향하라는게 있더라고요 

https://upcurvewave.tistory.com/101 이 글 참고해보시면 좋을 듯 합니다!

Reply:
감사합니다 제가 작성한 코드지만 좋은 코드라는 생각이 드는 코드는 아니라고 생각합니다.. 다음에는 주신 링크의 글을 참조해서 코드를 작성해보도록 하겠습니다 ! 
자세하게 변경할 부분의 예시까지 주셔서 감사합니다 !